Equella institutions can be configured to clash with admin.url

goetzk opened the pull request:
If an institution is imported using the same top level url as admin.url (nothing warns this is happening) visits to  "/institutions.do?method=admin" result in errors similar to the below.

```
Access denied
Sorry you do not have access to view the page you requested.
Only admin site access

The URL you requested was:
http://192.168.56.101/institutions.do?method=admin

Authentication:
You are not logged into the system

Action:
Go back
```

goetzk committed:
Explain that equella institutions can be configured to clash with admin.url

If an institution is imported using the same top level url as admin.url (nothing warns this is happening)
then visits to  "/institutions.do?method=admin" result in errors similar to the below.

```
Access denied
Sorry you do not have access to view the page you requested.
Only admin site access

The URL you requested was:
http://192.168.56.101/institutions.do?method=admin

Authentication:
You are not logged into the system

Action:
Go back
```

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -333,6 +333,8 @@ Once the server has been started, the success of the installation can be checked
 The openEQUELLA Server administration account is hidden from casual users and is displayed by entering a special URL created from the server’s base URL. 
 
+Note: The base URL is configured in `learningedge-config/mandatory-config.properties` as `admin.url`. If admin.url is changed adjust the instructions below accordingly.
+
```
I would change:

The openEQUELLA Server administration account is hidden from casual users and is displayed by entering a special URL created from the server’s base URL. 
Note: The base URL is configured in `learningedge-config/mandatory-config.properties` as `admin.url`. If admin.url is changed adjust the instructions below accordingly.

to

The openEQUELLA Server administration account is hidden from casual users and is displayed by entering a special URL.  This URL is configured in `learningedge-config/mandatory-config.properties` as `admin.url`.

goetzk replied:
I have applied the suggested wording in f3942e7

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -333,6 +333,8 @@ Once the server has been started, the success of the installation can be checked
+
 ### To open the Server administration account page
 1. Open a browser and enter the openEQUELLA address of the hosting server with 
 ‘/institutions.do?method=admin’ appended to the URL. (e.g. ‘http://equella.myinstitution.edu/logon.do’ would become ‘http://equella.myinstitute.edu/institutions.do?method=admin’).
```
I would change this to:

Open a browser and enter the openEQUELLA address of the `admin.url` with 
‘/institutions.do?method=admin’ appended.

cbeach47 replied:
:+1:

cbeach47 commented on guides/InstallationAdminGuide.md:
```diff
@@ -361,6 +363,8 @@ Server administrators are able to give institutions an arbitrary base URL. This
 **‘URL must not 'overwrite' an existing institution's URL space, in this case http://equella.myinstitution:4012/doco/qa2/. This may cause this institution to work incorrectly’.**
 
+Note: The first two examples above conflict with the default setting for `admin.url`. If using the top level for your repository change `admin.url` in `learningedge-config/mandatory-config.properties` to include `/adm` or similar.
+
```
I would change this to:

Note: The `admin.url` needs to have it's own URL space.  While possible to have your institution as the top level (my.oeq.org) and the `admin.url` be a context of it (my.oeq.org/adm/), it's not recommended.  A _cleaner_ approach would be `admin.my.oeq.org`.  A _more secure_ approach would be to set the `admin.url` to an IP address of the given server so it's only accessible behind your organization's firewall.

goetzk replied:
I've reworded that whole section in efbb2c4, I'd be interested to know what you think of the revised version.

On a related topic, should I open an issue against equella asking that admin be protected from institutions masking it just as other institutions are?

cbeach47 replied:
@goetzk - I would back off the use of  'recommended' for 
> Server administrators are able to give institutions an arbitrary base URL but it is **recommended** that they contain a context (`/something/` after the address)

As long as the system administrator is aware of the implications, a context is not needed for institutions, and multiple adopters use this flow.

Otherwise, looks good!

cbeach47 replied:
Sure, an issue ticket for /Equella works.  
> On a related topic, should I open an issue against equella asking that admin be protected from institutions masking it just as other institutions are?

goetzk replied:
Done as Equella/#675

goetzk replied:
Current wording says 'recommended' - so that is good to stay?

cbeach47 replied:
@goetzk - I would remove 'recommended'.

goetzk replied:
I've removed recommended.

cbeach47 reviewed (changes requested):
Thanks for flagging this @goetzk , it needed some revision.  I suggested some changes to clarify some points on the server admin URL.

goetzk committed:
Change wording around admin account location per feedback

The new text comes unchanged from @cbeach47's review.

goetzk committed:
Rework server administration access section

Based on feedback I started to rework the note about admin.url and it
snowballed in to fairly substantial changes to how admin.url is described.

goetzk commented:
@cbeach47 
I've pushed a couple of commits with fixes for feedback. I see the branch has developed a conflict with master; I'll make those fixes when we're happy with the rest of the changes.

cbeach47 commented:
@goetzk - minor changes requested, and please resolve the conflicts - I'll keep a closer eye on this thread so it doesn't age.

goetzk committed:
Merge remote-tracking branch 'upstream/master' into document-admin-institution-clashes

goetzk commented:
@cbeach47 rebased to master

goetzk committed:
Downgrade recommended to suggested

Last of the changes requred by @cbeach47.

goetzk commented:
@cbeach47 all changes made.
